connaisseur/validators/cosign/cosign_validator.py

```diff
@@ -280,9 +299,11 @@ def __invoke_cosign(self, image: str, tr_args: dict):
 
         Raises an exception if Cosign times out.
         """
-        cmd = [
-            "/app/cosign/cosign",
+
+        args = [
             "verify",
+            *([] if verify_transparency_log else ["--insecure-ignore-tlog"]),
+            "--insecure-ignore-sct",
```

Comment:
excellent question! answer: i don't know. without setting this to `--insecure-ignore-sct` the async execution will fail, because everyone tries to access a certain file, in a locking manner -> only one succeed, the rest says: resource locked.

now what does this setting exactly do? here the `--help` text: 
```
when set, verification will not check that a certificate contains an embedded SCT, a proof of inclusion in a certificate transparency log
``` 
i don't really know what this implies and further research did not lead to anything :( all i know is, without setting this, everything breaks.

Reply:
That doesn't sound good though. We shouldn't be setting insecure flags just "to make it work". A workaround would be sequential operations (slow but secure) or you could add an insecure flag in our config that you need to set to use Cosign or you could report this as an issue upstream and see what falls out...

Reply:
So I looked into it again. SCT stands for, I assume, Signed Certificate Timestamp (never explained by sigstore/cosign). It's a proof that a certificate was added to a transparency log. The thing is, we currently don't use any certificates regarding the use of cosign.

There is a usecase for certificates when talking about cosign. That is when doing keyless signatures and the verification of those. In this case cosign generates an ephemeral certificate using fulcio (another project from sigstore) which is then used to do signatures with. This certificate can be added to the tranparency log and the SCT-flag can do additional verifications that the certificates actually does reside in the log.

For now I would skip all of this, since we currently don't support keyless signatures anyways and have no certificates for verify for. But it will almost definately be a topic once we decide to support keyless and it will probably a major pain 😢

Reply:
Would add a comment explaining this to the code. Also would then link that code to the #407 issue to make sure this isn't forgotten if we implement keyless someday

Reply:
added a comment.